src/main/java/frc/robot/subsystems/CoralManipulator.java

```diff
@@ -0,0 +1,147 @@
+package frc.robot.subsystems;
+
+import static edu.wpi.first.units.Units.*;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.DoublePublisher;
+import edu.wpi.first.networktables.BooleanPublisher;
+import edu.wpi.first.units.AngleUnit;
+import edu.wpi.first.units.CurrentUnit;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.VoltageUnit;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import frc.util.control.nt.PIDTuning;
+import frc.util.control.nt.TunableNumber;
+
+import java.util.function.BooleanSupplier;
+
+public class CoralManipulator extends SubsystemBase {
+    private final TalonFX pivotMotor;
+    private final TalonFX rollerMotor;
+    private final DigitalInput beambreak;
+    private final MotionMagicVoltage motionMagic;
+    
+    private final TunableNumber kP;
+    private final TunableNumber kI;
+    private final TunableNumber kD;
+    private final PIDTuning pidTuning;
+
+    private final NetworkTable manipulatorTable;
+    private final DoublePublisher anglePub;
+    private final BooleanPublisher beambreakPub;
+    private final DoublePublisher currentPub;
+    
+    private boolean openLoop;
+    private Measure<AngleUnit> targetAngle;
+
+    private static final Measure<CurrentUnit> CURRENT_LIMIT = Amps.of(35);
+
+    public CoralManipulator() {
+        pivotMotor = new TalonFX(Constants.Pivot.PIVOT_MOTOR_ID);
+        rollerMotor = new TalonFX(Constants.Pivot.ROLLER_MOTOR_ID);
```

Comment:
don't use a motor config in constants if you're just going to create the talon's directly